### PR TITLE
Feature/bookmark functionality

### PR DESCRIPTION
**Pull request by CorneliusMoe**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -6,16 +6,22 @@
...
   <body>
     <header>
       <h1><a href="#">Quiz App</a></h1>
     </header>
     <main class="main-flex">
       <section class="questionbox">
-        <img src="assets/images/bookmark_dark.svg" class="bookmark" />
+        <img
+          src="assets/images/bookmark_dark.svg"
+          class="bookmark"
```
Comment (FelineHuhn) [inline index.html:19]:
```suggestion
          class="questionbox__bookmark"
```

Comment (FelineHuhn) [reply index.html:19]: When we want to name our css classes with the BEM Method, it should have a classname like this.

File: index.html [hunk 2]
```diff
@@ -6,16 +6,22 @@
...
+          data-js="bookmarkIcon"
+        />
         <p class="question">What property flips the axes in flexbox?</p>
```
Comment (FelineHuhn) [inline index.html:22]:
```suggestion
        <p class="questionbox__question">What property flips the axes in flexbox?</p>
```

Comment (FelineHuhn) [reply index.html:22]: same here, see description above (BEM)

File: index.html [hunk 3]
```diff
@@ -6,16 +6,22 @@
...
-        <button type="button">Show Answer</button>
+        <button type="button" data-js="button">Show Answer</button>
```
Comment (FelineHuhn) [inline index.html:23]:
```suggestion
        <button class="questionbox__button-answer" type="button" data-js="button">Show Answer</button>
```

Comment (FelineHuhn) [reply index.html:23]: same here, see description above (BEM)

File: index.html [hunk 4]
```diff
@@ -6,16 +6,22 @@
...
+        <p class="answer answer--hidden" data-js="answer">flex-direction</p>
```
Comment (FelineHuhn) [inline index.html:24]:
```suggestion
        <p class="questionbox__answer questionbox__answer--hidden" data-js="answer">flex-direction</p>
```

Comment (FelineHuhn) [reply index.html:24]: same here, see description above (BEM)

File: index.html [hunk 5]
```diff
@@ -6,16 +6,22 @@
...
         <ul class="tags">
```
Comment (FelineHuhn) [inline index.html:25]:
```suggestion
        <ul class="questionbox__tag-list">
```

Comment (FelineHuhn) [reply index.html:25]: same here, see description above (BEM)

File: index.html [hunk 6]
```diff
@@ -6,16 +6,22 @@
...
           <li class="tag">#html</li>
```
Comment (FelineHuhn) [inline index.html:26]:
```suggestion
          <li class="questionbox__tag-list-item">#html</li>
```

File: index.html [hunk 7]
```diff
@@ -6,16 +6,22 @@
...
           <li class="tag">#flexbox</li>
```
Comment (FelineHuhn) [inline index.html:27]:
```suggestion
          <li class="questionbox__tag-list-item">#flexbox</li>
```

File: index.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+const bookmarkIcon = document.querySelector('[data-js="bookmarkIcon"]');
+const button = document.querySelector('[data-js="button"]');
+const answer = document.querySelector('[data-js="answer"]');
+
+bookmarkIcon.addEventListener("click", () => {
+  let inactive = "/assets/images/bookmark_dark.svg";
```
Comment (FelineHuhn) [inline index.js:6]:
```suggestion
  const inactive = "/assets/images/bookmark_dark.svg";
```

File: index.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  let active = "/assets/images/bookmark_light.svg";
```
Comment (FelineHuhn) [inline index.js:7]:
```suggestion
  const active = "/assets/images/bookmark_light.svg";
```

File: index.js (hunk 8 above)
Comment (FelineHuhn) [reply index.js:6]: Since the variable is not changed, we should use `const` here.

File: index.js (hunk 9 above)
Comment (FelineHuhn) [reply index.js:7]: Same here, see above :)

File: index.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+  if (bookmarkIcon.getAttribute("src") === inactive) {
```
Comment (FelineHuhn) [inline index.js]: This works  :) But technically, it's not a "real" toggle function. You could create a `button` element wrapped around your `img` tag and could create a css class (something like "questionbox__bookmark--active" with a coloring style and than toggle this class with the syntax `bookmarkIcon.classList.toggle("questionbox__bookmark--active")`

File: index.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+const bookmarkIcon = document.querySelector('[data-js="bookmarkIcon"]');
+const button = document.querySelector('[data-js="button"]');
```
Comment (FelineHuhn) [inline index.js:2]:
```suggestion
const answerButton = document.querySelector('[data-js="button"]');
```

Comment (FelineHuhn) [reply index.js:2]: This button variable maybe should have a more descriptive name, or?

File: index.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  let active = "/assets/images/bookmark_light.svg";
+
+  if (bookmarkIcon.getAttribute("src") === inactive) {
+    bookmarkIcon.setAttribute("src", active);
+  } else {
+    bookmarkIcon.setAttribute("src", inactive);
+  }
+});
+
+button.addEventListener("click", () => {
```
Comment (FelineHuhn) [inline index.js:15]:
```suggestion
answerButton.addEventListener("click", () => {
```

File: index.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  answer.classList.toggle("answer--hidden");
+  if (button.textContent === "Hide Answer") {
```
Comment (FelineHuhn) [inline index.js:17]:
```suggestion
  if (answerButton.textContent === "Hide Answer") {
```

File: index.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+    button.textContent = "Show Answer";
```
Comment (FelineHuhn) [inline index.js:18]:
```suggestion
    answerButton.textContent = "Show Answer";
```

File: index.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+  } else {
+    button.textContent = "Hide Answer";
```
Comment (FelineHuhn) [inline index.js:20]:
```suggestion
    answerButton.textContent = "Hide Answer";
```

File: css/styles.css [hunk 16]
```diff
@@ -79,6 +79,16 @@ main {
...
+  text-align: center;
+  font-weight: bold;
+  color: antiquewhite;
+}
+
+.answer--hidden {
+  display: none;
+}
+
 button {
```
Comment (FelineHuhn) [inline css/styles.css:92]:
```suggestion
.questionbox__button-answer {
```

**Review by FelineHuhn — CHANGES_REQUESTED** [review]
You did a really great job so far Cornelius! 🥳 I left you some comments with a few little suggestions. I was not able to mark everything in the lines, but i hope, you get an idea of the BEM method to name your css classes in a more descriptive way. 

Keep up the good work! 🚀

**Comment by CorneliusMoe** [comment]
@FelineHuhn Thank you for the in-depth code review. I made changes to my classes so BEM is applied. I also changed my index.js according to your recommendations.

**Pull request merged**